src/restLink.ts

```diff
@@ -269,15 +350,53 @@ const resolver: Resolver = async (
         'Missing params to run query, specify it in the query params or use an export directive',
       );
     }
-    let { method, type } = directives.rest;
+    let {
+      method,
+      type,
+      bodyBuilder,
+      bodyKey,
+      fieldNameDenormalizer: perRequestNameDenormalizer,
+    } = directives.rest as RestLink.DirectiveOptions;
     if (!method) {
       method = 'GET';
     }
-    validateRequestMethodForOperationType(method, 'query');
+
+    let body = null;
+    if (
+      -1 === ['GET', 'DELETE'].indexOf(method) &&
```

Comment:
This is a tiny detail but now that we have [includes method](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/includes), I have found it to be clearer that indexOf

Reply:
I had to polyfill it in my project, last week -- are you sure it's safe to use in all browsers / babel / web pack / typescript environments?

Reply:
You are right, I have a polyfill in my setup also, let's keep it that way

Reply:
maybe premature optimization but maybe the arrays containing the types should be replaced with method `method === 'GET' || method === 'DELETE'` to prevent recreating the arrays on every function invocation? I also did see this in other methods. Alternatively the arrays containing these types could be lifted up in the scoe

Reply:
Since there are only 2 distinct calls that look like this, and they have different values, I do agree it's probably a premature optimization. These calls would happen once each per network call, and that's just about the slowest thing in the world anyhow ;).